dapps/assets/src/components/contacts/ContactsService.ts

```diff
@@ -0,0 +1,103 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+import * as bcc from '@evan.network/api-blockchain-core';
+import InviteDispatcher from './InviteDispatcher';
+
+export interface Contact {
+  address: string;
+  alias: string;
+  createdAt: string;
+  favorite: boolean;
+  icon: string;
+  type: ContactType;
+  updatedAt: string;
+}
+
+enum ContactType {
+  COMPANY = 'company',
+  IOT_DEVICE = 'device',
+  USER = 'user',
+  UNSHARED = 'unshared'
+}
+
+export class ContactsService {
+  private contacts;
+  private runtime;
+
+  constructor(runtime) {
+    this.runtime = runtime;
+  }
+
+  async getContacts(): Promise<Contact[]> {
+    this.contacts = await this.runtime.profile.getAddressBook();
+
+    let data: Contact[] = [];
+    Object.keys(this.contacts.profile).forEach(async contact => {
+      const type = await this.getProfileType(contact);
+      data.push({
+        address: contact,
+        alias: this.contacts.profile[contact].alias,
+        createdAt: this.contacts.profile[contact].createdAt,
+        favorite: Math.random() > 0.5 ? true : false, // TODO: hard coded
```

Comment:
Should be loaded and saved within the contact.

Reply:
What exactly do you mean by that?

Reply:
I mean, we should implement the save and loading logic for the favorite flag for the contacts^^